e2e/README.md

```diff
@@ -0,0 +1,81 @@
+# End-to-end tests
+
+This directory contains end-to-end (i.e. browser based) tests for Shiny for Python.
+
+The Python files directly in this subdirectory are for Pytest fixtures and helper code
+to make test writing easier. (Eventually this logic may move to the `shiny` package
+itself or its own dedicated package, so that Shiny app authors can set up their own e2e
+tests against their apps.)
+
+The actual tests are in subdirectories. Each subdirectory contains one or more Pytest
+files (`test_*.py`) containing [Playwright](https://playwright.dev/python/) assertions,
+and optionally, a single app (`app.py`) that the assertions test against. (The app is
+optional, because the tests may also be for apps in the `examples/` directory.)
+
+## Running tests
+
+The following commands can be run from the repo root:
+
+```sh
+# Run all e2e tests
+make e2e
+# Another way to run all e2e tests
+tox
+
+# Run just the tests in e2e/async/
+tox e2e/async
+
+# Run just the tests in e2e/async/, in headed mode
+tox -- --headed e2e/async
+```
+
+## Shiny app fixtures
+
+Playwright for Python launches and controls (headless or headful) browsers, but does not
+know how to start/stop Shiny app processes. Insteaed, we have our own [Pytest
+fixtures](https://docs.pytest.org/en/latest/explanation/fixtures.html) for
+Shiny app process management.
+
+### Testing a local `app.py`
+
+The `local_app: ShinyAppProc` fixture launches the `app.py` in the current directory
+as the calling `test_*.py` file.
+
+```python
+# pyright: reportUnknownMemberType=false
+
+import re
+from playwright.sync_api import Page, expect
+from conftest import ShinyAppProc
```

Comment:
I'm not sure what the full implication of a relative import might be here, but to me it makes it more clear where the conftest module lives? 

```suggestion
from .conftest import ShinyAppProc
```

Reply:
`conftest` wouldn't be in the same directory as a test, but in the parent directory. So it'd be `..conftest`. But I'm not sure that would work because the e2e stuff isn't in a package.

Reply:
Yeah, just tried it, it indeed does not work.